src/hypergol/base_tensorflow_model_block.py

```diff
@@ -0,0 +1,37 @@
+# pylint: disable=E0611, W0235
+
+import inspect
+import json
+from tensorflow.python.keras import layers
+
+
+class BaseTensorflowModelBlock(layers.Layer):
+    """Subclasses tensorflow-keras layers to provide logical groupings of functionality/layers."""
+
+    def __init__(self, **kwargs):
+        super(BaseTensorflowModelBlock, self).__init__(**kwargs)
+
+    def get_config(self):
+        parameters = {k: v for k, v in self.__dict__.items() if k in inspect.signature(self.__class__).parameters.keys()}
+        config = super(BaseTensorflowModelBlock, self).get_config()
+        for name, value in parameters.items():
+            config.update({name: value})
```

Comment:
why not ```config[name]=value```?

Reply:
actually can we make this explicit?
```
config = super(BaseTensorflowModelBlock, self).get_config()
for name, value in self.__dict__.items():
     if name in inspect.signature(self.__class__).parameters.keys():
        config[name] = value
return config
```

Reply:
maybe this can be extracted as `constructorParameters=inspect.signature(self.__class__).parameters.keys()`